### ISSUE TITLE
Feature Request: Add interface to easily add SMTP server connection strings

### ISSUE THREAD

**Issue by DBailey635** [Orckestra/C1-CMS#147]
In a similar way to adding a SQL database connection string, please could you add a user interface for adding SMTP email server connection strings?  Additionally, it may be useful to add these fields to the initial system set-up form.

[Current method](http://docs.composite.net/Configuration/Configuring-Email-Host)

This would help in situations where C1 is pre-installed by a web hosting company, and where admins lack the technical knowledge or confidence to change the web.config file.

**Comment by wwwysocki** [comment]
+

Just FYI: Some packages like Newsletter or Extranet add some default mail settings in web.config when installed. Something like this:

```
<system.net>
  <mailSettings>
    <smtp deliveryMethod="Network">
      <network host="localhost" />
    </smtp>
  </mailSettings>
</system.net>
```

The Venus starter site comes with the settings above, too. Added by some package - not sure which.

**Comment by DBailey635** [comment]
Thanks @wwwysocki . I was suggesting adding this to the core of C1, so that add-ins wouldn't need to specify their own mechanisms to make emailing work.

**Comment by burningice2866** [comment]
@DBailey635 What about just letting this new field in the setup-form write to system.net > mailSettings > smtp > network in web.config directly?

As long as emails are being sent using the standard SmtpClient in .Net without any argument, this is the only place the smtp host can be configured.

Otherwise look at using CompositeC1Contrib.Email which is a complete email infrastructure for C1 where can define all smtp-related settings directly from within the Console.

**Comment by DBailey635** [comment]
@burningice2866 That's exactly how I'd do it initially, but sometimes the SMTP strings need to be altered (password changes, for example), so having a UI in C1 Core to do so after set-up would also be beneficial.

**Comment by burningice2866** [comment]
@DBailey635 i can make a web.config editor package, with tabs for various often-used sections and simple textboxes, checkboxes and dropdowns. Would that work?

**Comment by DBailey635** [comment]
Thanks for the offer @burningice2866. As a short-term solution that might be a good idea, but I think adding this to C1 Core should be the long-term goal for the reasons outlined above.

**Comment by burningice2866** [comment]
@DBailey635 in such case is it easy to name many handfuls of web.config settings which could be exposed through the Console. Where do you set the limit, who decides what and what not should be easy configurable?

Not even the sql connection string is configurable through the console, that also requires manual edit of web.config.

**Comment by DBailey635** [comment]
@burningice2866 Yes, indeed. However, this request is just for an SMTP connection string UI to enable the email functionality specified in issues #148, #149, as well as the add-ons mentioned above and others that also send email messages.

**Comment by burningice2866** [comment]
@DBailey635 i know, its just hard to justify adding a whole new interface for one web.config setting, when there are so many other settings which also could benefit being edited from the Console.

Btw, IIS Manager has a nice UI for editing most parts of web.config http://mscerts.wmlcloud.com/image/201012/Configuring%20SMTP_12.jpg
